beep/structure/biologic.py

```diff
@@ -119,14 +119,16 @@ def from_file(cls, path, mapping_file=None):
         data["data_point"] = list(range(1, len(raw["cycle number"]) + 1))
 
         data = pd.DataFrame(data)
-        data.loc[data.step_index % 2 == 0, "charge_capacity"] = abs(
-            data.charge_capacity
-        )
-        data.loc[data.step_index % 2 == 1, "charge_capacity"] = 0
-        data.loc[data.step_index % 2 == 1, "discharge_capacity"] = abs(
-            data.discharge_capacity
-        )
-        data.loc[data.step_index % 2 == 0, "discharge_capacity"] = 0
+        # not applicable to .txt -> check raw data formats before running the following conversion
+        if os.path.splitext(path)[1] == ".csv":
```

Comment:
Is there a reason that the csv file will be different than the txt file?

Reply:
Now txt and csv files are treated the same. A test was added for a new txt file.